### ISSUE TITLE
The column does not exist in the current database.

### ISSUE THREAD

**Issue by baglianiinovaetech** [prisma/prisma#22098]
### Bug description
When executing find on the schema below it returns an error
The column `TESTE_NÃMERICO` does not exist in the current database.

In versions <= 5.3.0 it works

### How to reproduce
Clone `https://github.com/baglianiinovaetech/ErrorMapPrisma` and execute start

Or

Used example "https://www.prisma.io/docs/getting-started/quickstart" with the schema below

Create file script.ts

```ts
import { PrismaClient } from "@prisma/client";

const prisma = new PrismaClient();

async function main() {
  await prisma.test.findFirst();
}

main()
  .then(async () => {
    await prisma.$disconnect();
  })
  .catch(async (e) => {
    console.error(e);
    await prisma.$disconnect();
    process.exit(1);
  });
```

run npx ts-node script.ts

The column `TESTE_NÃMERICO` does not exist in the current database.

### Prisma information

```prisma
generator client {
  provider = "prisma-client-js"
}

datasource db {
  provider = "sqlserver"
  url      = env("DATABASE_URL")
}

model test {
    id             Int    @default(autoincrement())
    TESTE_N_MERICO String @unique @map("TESTE_NÚMERICO")
}
```
```ts
import { PrismaClient } from "@prisma/client";

const prisma = new PrismaClient();

async function main() {
  await prisma.test.findFirst({});
}

main()
  .then(async () => {
    await prisma.$disconnect();
  })
  .catch(async (e) => {
    console.error(e);
    await prisma.$disconnect();
    process.exit(1);
  });

```

### Environment & setup

- OS: Windows 10
- Database:  SQL Server
- Node.js version: v18.17.1
- Yarn version: v1.22.19

### Prisma Version

```
 "@prisma/client": "^5.6.0",
 "prisma": "^5.6.0",
```

**Comment by SevInf** [comment]
Can confirm that it works on `5.2.0` and does not on `5.3.0`-`5.7.0.dev.45`

**Comment by baglianiinovaetech** [comment]
v5.7.0-dev.45

```cmd
$ yarn start
yarn run v1.22.19
$ yarn install && npx prisma generate & npx prisma db push && npx ts-node script.ts
[1/4] Resolving packages...
success Already up-to-date.
Environment variables loaded from .env
Prisma schema loaded from prisma\schema.prisma

✔ Generated Prisma Client (v5.7.0-dev.45) to .\node_modules\@prisma\client in 75ms
Start using Prisma Client in Node.js (See: https://pris.ly/d/client)

import { PrismaClient } from '@prisma/client'
const prisma = new PrismaClient()

or start using Prisma Client at the edge (See: https://pris.ly/d/accelerate)

import { PrismaClient } from '@prisma/client/edge'
const prisma = new PrismaClient()


See other ways of importing Prisma Client: http://pris.ly/d/importing-client


warn Versions of prisma@5.2.0 and @prisma/client@5.7.0-dev.45 don't match.
This might lead to unexpected behavior.
Please make sure they have the same version.
Environment variables loaded from .env
Prisma schema loaded from prisma\schema.prisma
Datasource "db": SQL Server database

The database is already in sync with the Prisma schema.

✔ Generated Prisma Client (v5.7.0-dev.45) to .\node_modules\@prisma\client in 93ms

PrismaClientKnownRequestError: 
Invalid `prisma.test.findMany()` invocation in
C:\api\ErrorMapPrisma\script.ts:6:21

  3 const prisma = new PrismaClient();
  4
  5 async function main() {
→ 6   await prisma.test.findMany(
The column `TESTE_NÃMERICO` does not exist in the current database.
    at ii.handleRequestError (C:\api\ErrorMapPrisma\node_modules\@prisma\client\runtime\library.js:124:6817)
    at ii.handleAndLogRequestError (C:\api\ErrorMapPrisma\node_modules\@prisma\client\runtime\library.js:124:6206)
    at ii.request (C:\api\ErrorMapPrisma\node_modules\@prisma\client\runtime\library.js:124:5926)
    at async l (C:\api\ErrorMapPrisma\node_modules\@prisma\client\runtime\library.js:129:10023) {
  code: 'P2022',
  clientVersion: '5.7.0-dev.45',
  meta: { column: 'TESTE_NÃ\x9AMERICO' }
}
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```


v5.2.0
There was no error

```cmd
$ yarn start
yarn run v1.22.19
$ yarn install && npx prisma generate & npx prisma db push && npx ts-node script.ts
[1/4] Resolving packages...
success Already up-to-date.
Environment variables loaded from .env
Prisma schema loaded from prisma\schema.prisma

✔ Generated Prisma Client (v5.2.0) to .\node_modules\@prisma\client in 76ms
Start using Prisma Client in Node.js (See: https://pris.ly/d/client)

import { PrismaClient } from '@prisma/client'
const prisma = new PrismaClient()

or start using Prisma Client at the edge (See: https://pris.ly/d/accelerate)

import { PrismaClient } from '@prisma/client/edge'
const prisma = new PrismaClient()

See other ways of importing Prisma Client: http://pris.ly/d/importing-client

Environment variables loaded from .env
Prisma schema loaded from prisma\schema.prisma
Datasource "db": SQL Server database

The database is already in sync with the Prisma schema.

✔ Generated Prisma Client (v5.2.0) to .\node_modules\@prisma\client in 90ms

┌─────────────────────────────────────────────────────────┐
│  Update available 5.2.0 -> 5.6.0                        │
│  Run the following to update                            │
│    yarn add --dev prisma@latest                         │
│    yarn add @prisma/client@latest                       │
└─────────────────────────────────────────────────────────┘
Done in 15.06s.
```

**Comment by baglianiinovaetech** [comment]
v5.3.0 it works

```
$ yarn start
yarn run v1.22.19
$ yarn install && npx prisma generate & npx prisma db push && npx ts-node script.ts
[1/4] Resolving packages...
success Already up-to-date.
Environment variables loaded from .env
Prisma schema loaded from prisma\schema.prisma

✔ Generated Prisma Client (v5.3.0) to .\node_modules\@prisma\client in 82ms

Start using Prisma Client in Node.js (See: https://pris.ly/d/client)

import { PrismaClient } from '@prisma/client'
const prisma = new PrismaClient()

or start using Prisma Client at the edge (See: https://pris.ly/d/accelerate)

import { PrismaClient } from '@prisma/client/edge'
const prisma = new PrismaClient()


See other ways of importing Prisma Client: http://pris.ly/d/importing-client

Environment variables loaded from .env
Prisma schema loaded from prisma\schema.prisma
Datasource "db": SQL Server database

The database is already in sync with the Prisma schema.

✔ Generated Prisma Client (v5.3.0) to .\node_modules\@prisma\client in 96ms

┌─────────────────────────────────────────────────────────┐
│  Update available 5.3.0 -> 5.6.0                        │
│  Run the following to update                            │
│    yarn add --dev prisma@latest                         │
│    yarn add @prisma/client@latest                       │
└─────────────────────────────────────────────────────────┘
Done in 19.48s.
```

**Comment by baglianiinovaetech** [comment]
I will check which specific version works and which one stopped working.

**Comment by baglianiinovaetech** [comment]
v5.5.2 it works
```
$ yarn start
yarn run v1.22.19
$ yarn install && npx prisma generate & npx prisma db push && yarn ts-node script.ts
[1/4] Resolving packages...
success Already up-to-date.
Environment variables loaded from .env
Prisma schema loaded from prisma\schema.prisma

✔ Generated Prisma Client (v5.5.2) to .\node_modules\@prisma\client in 113ms

Start using Prisma Client in Node.js (See: https://pris.ly/d/client)
```
import { PrismaClient } from '@prisma/client'
const prisma = new PrismaClient()
```
or start using Prisma Client at the edge (See: https://pris.ly/d/accelerate)
```
import { PrismaClient } from '@prisma/client/edge'
const prisma = new PrismaClient()
```

See other ways of importing Prisma Client: http://pris.ly/d/importing-client

┌─────────────────────────────────────────────────────────┐
│  Update available 5.5.2 -> 5.6.0                        │
│  Run the following to update                            │
│    yarn add --dev prisma@latest                         │
│    yarn add @prisma/client@latest                       │
└─────────────────────────────────────────────────────────┘
Environment variables loaded from .env
Prisma schema loaded from prisma\schema.prisma
Datasource "db": SQL Server database

The database is already in sync with the Prisma schema.

✔ Generated Prisma Client (v5.5.2) to .\node_modules\@prisma\client in 98ms

$ C:\api\ErrorMapPrisma\node_modules\.bin\ts-node script.ts
EXECUTED
Done in 13.88s.
```

**Comment by Jolg42** [comment]
Note: @Druue mentioned in https://github.com/prisma/prisma/pull/23242 which adds a regression test that it was already fixed in `5.10`.

@baglianiinovaetech Did you try `5.10.0` already? Does it work for you?

**Comment by baglianiinovaetech** [comment]
`5.10.0` it work, thanks 🎉
